### PR TITLE
Update libSDL package and add X and Wayland bindings

### PR DESCRIPTION
**Pull request by JL2210**
Needed for SDL plugins, such as SDL_image and SDL_net.

### CODE REVIEW

File: packages/libsdl.rb [hunk 1]
```diff
@@ -3,32 +3,31 @@
...
   })
   binary_sha256 ({
-    aarch64: 'dd572e285fc4485ec46c36f7612f06f78ee2565e4fa5814c2828c520ea7f44ac',
-     armv7l: 'dd572e285fc4485ec46c36f7612f06f78ee2565e4fa5814c2828c520ea7f44ac',
-       i686: '485362d999a6a6ab466c8a544df175b9e93db2b5149ee0f37be4bf33d6adc66b',
-     x86_64: 'f9c9cee413c6acd449fefadb44f8148595124d994b658fedca67642921369b43',
   })
 
   depends_on 'dbus'
   depends_on 'eudev'
```
Comment (uberhacker) [inline packages/libsdl.rb]: `xwayland` already has `eudev` as a dependency.

File: packages/libsdl.rb [hunk 2]
```diff
@@ -3,32 +3,31 @@
...
-     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/libsdl-1.2.15-2-chromeos-x86_64.tar.xz',
   })
   binary_sha256 ({
-    aarch64: 'dd572e285fc4485ec46c36f7612f06f78ee2565e4fa5814c2828c520ea7f44ac',
-     armv7l: 'dd572e285fc4485ec46c36f7612f06f78ee2565e4fa5814c2828c520ea7f44ac',
-       i686: '485362d999a6a6ab466c8a544df175b9e93db2b5149ee0f37be4bf33d6adc66b',
-     x86_64: 'f9c9cee413c6acd449fefadb44f8148595124d994b658fedca67642921369b43',
   })
 
   depends_on 'dbus'
```
Comment (uberhacker) [inline packages/libsdl.rb]: `pulseaudio` already has `dbus` as a dependency.

File: packages/libsdl.rb [hunk 3]
```diff
@@ -3,32 +3,31 @@
...
   depends_on 'eudev'
+  depends_on 'xwayland'
+  depends_on 'alsa_lib'
+  depends_on 'pulseaudio'
+  depends_on 'mesa'
+  depends_on 'nasm' => :build
+
+  def self.patch
+    system "sed -i 's,#ifdef LONG64,#if 0 // Causes a build failure with recent libx11 versions,g' src/video/x11/SDL_x11sym.h"
```
Comment (uberhacker) [inline packages/libsdl.rb]: If you do this, do you realize everything inside this block will get ignored?

Comment (JL2210) [reply packages/libsdl.rb]: It's only three statements. They all conflict with `libx11`.

**Comment by JL2210** [comment]
@uberhacker Not a duplicate. Two different packages.

File: packages/libsdl.rb (hunk 2 above)
Comment (JL2210) [reply packages/libsdl.rb]: Done.

File: packages/libsdl.rb (hunk 1 above)
Comment (JL2210) [reply packages/libsdl.rb]: Done.

File: packages/libsdl.rb (hunk 3 above)
Comment (JL2210) [reply packages/libsdl.rb]: Conflicting statements:
```
/*
 * Not required...these only exist in code in headers on some 64-bit platforms,
 *  and are removed via macros elsewhere, so it's safe for them to be missing.
 */
#ifdef LONG64
SDL_X11_MODULE(IO_32BIT)
SDL_X11_SYM(int,_XData32,(Display *dpy,register long *data,unsigned len),(dpy,data,len),return)
SDL_X11_SYM(void,_XRead32,(Display *dpy,register long *data,long len),(dpy,data,len),)
#endif
```

**Pull request merged**